### PR TITLE
Feature/add number of tokens to inst config

### PR DESCRIPTION
**Pull request by MKodde**
**Test notes**
* The commit messages talk you through the changes.
* Please check if I got the database schema correctly in sync with the default values configured in the application.
* The postman 2.json config was updated partly to use the new dev-vm settings (hostnames/secrets changed). I suggest opening a new PR to fix this in its entirety.

### CODE REVIEW

File: app/DoctrineMigrations/Version20180409100948.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE institution_configuration_options ADD number_of_tokens_per_identity_option TINYINT(1) DEFAULT \'1\' NOT NULL');
```
[the remark this answers is not in the document]
Comment (jorissteyn) [reply app/DoctrineMigrations/Version20180409100948.php]: Another thought: if MW expects the default value to be '1', but existing institutions have a value of '0', doesn't that mean the first push to the institution configuration might not update the option correctly? Just like the bug with 'verify_email'?

[the remark this answers is not in the document]
Comment (MKodde) [reply app/DoctrineMigrations/Version20180409100948.php]: 1. Thanks for spotting the bug. The default value was set to the incorrect value in the ALTER TABLE query.
2. Not entirely right: The default for new institutions is the parameters.yml configured number of tokens (which is 3) IF the `number_of_tokens_per_identity` is set to 0 (the DISABLED value). Does this make sense?
3. Exactly that bug was fixed after implementing the feedback from your first review entry in this thread.

File: app/config/parameters.yml.dist [hunk 2]
```diff
@@ -50,5 +50,10 @@ parameters:
...
         tiqr:
             loa: 3
-    # Sets the number of tokens allowed for each identity
+
+    # Sets the number of tokens allowed for each identity.
+    #
+    # This is the global, application wide default. This configuration should be specified for each institution in the
+    # institution config (middleware api). The value configured in the parameters.yml will be used as the
+    # fallback/default value.
     number_of_tokens_per_identity: 3
```
[the remark this answers is not in the document]
Comment (MKodde) [reply app/config/parameters.yml.dist]: Three has been the default from the start. So for now I've left it like that. I will open a discussion in the Pivotal story to verify the correct setting.

[the remark this answers is not in the document]
Comment (MKodde) [reply app/config/parameters.yml.dist]: I don't realy get:

> OpenConext/Stepup-Middleware-clientbundle#57 allows only two tokens.

Can you be more specific? Or are you aiming at the unit test where the `number_of_tokens_per_identity` in the DTO is set to 2?

File: src/Surfnet/Stepup/Configuration/Value/NumberOfTokensPerIdentityOption.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        return new self(self::DISABLED);
+    }
+
+    public function __construct($numberOfTokensPerIdentity)
+    {
+        if (!is_numeric($numberOfTokensPerIdentity)) {
+            throw InvalidArgumentException::invalidType(
+                'integer',
+                'numberOfTokesnPerIdentity',
```
[the remark this answers is not in the document]
Comment (MKodde) [reply src/Surfnet/Stepup/Configuration/Value/NumberOfTokensPerIdentityOption.php]: Fixed

File: app/config/parameters.yml.dist (hunk 2 above)
[the remark this answers is not in the document]
Comment (jorissteyn) [reply app/config/parameters.yml.dist]: I was looking at the test case, never mind! As discussed: update query should be removed.

**Pull request merged**